dcap/ql/src/quote_enclave.rs

```diff
@@ -8,17 +8,167 @@
 //! has a mix up.  It uses the *verification* description for `sgx_ql_set_path`
 //! and the "generation" description for `sgx_qv_set_path`
 
+use crate::Error;
+use core::result::Result as CoreResult;
 use mc_sgx_core_sys_types::sgx_target_info_t;
 use mc_sgx_core_types::TargetInfo;
 use mc_sgx_dcap_ql_types::PathKind;
-use mc_sgx_dcap_types::{Quote3Error, RequestPolicy};
+use mc_sgx_dcap_types::RequestPolicy;
 use mc_sgx_util::ResultInto;
-use std::{ffi::CString, os::unix::ffi::OsStrExt, path::Path};
+use once_cell::sync::Lazy;
+use std::{ffi::CString, os::unix::ffi::OsStrExt, path::Path, sync::Mutex};
+
+/// A convenience type alias for a `Result` which contains an [`Error`].
+pub type Result<T> = CoreResult<T, Error>;
+
+/// Initialization of the paths for the quoting enclaves and quote provider
+/// library
+///
+/// This should only be called once during process start up utilizing
+/// [`PathInitializer::with_paths()`] or [`PathInitializer::try_default()`].
+/// If a consumer of this crate does not explicitly initialize the paths, then
+/// they will be defaulted on the first call to an SGX function that needs the
+/// paths set.
+#[derive(Debug)]
+pub struct PathInitializer;
+
+static PATH_INITIALIZER: Lazy<Mutex<Option<PathInitializer>>> = Lazy::new(|| Mutex::new(None));
+
+impl PathInitializer {
```

Comment:
@jcape in discussions yesterday, you mentioned an inner type.  I don't think it's necessary, bringing it up in case I missed something and to communicate that I didn't ignore or forget the comment.

Reply:
no barks resolving